Dockerfile

```diff
@@ -24,33 +27,48 @@ LABEL maintainer="Microsoft" \
 
 WORKDIR azure-cli
 COPY . /azure-cli
-# pip wheel - required for CLI packaging
-# jmespath-terminal - we include jpterm as a useful tool
-RUN pip install --no-cache-dir --upgrade pip wheel jmespath-terminal
+
 # bash gcc make openssl-dev libffi-dev musl-dev - dependencies required for CLI
-# jq - we include jq as a useful tool
 # openssh - included for ssh-keygen
 # ca-certificates
 # wget - required for installing jp
-RUN apk add --no-cache bash gcc make openssl-dev libffi-dev musl-dev jq openssh \
-    ca-certificates wget openssl git && update-ca-certificates
-# We also, install jp
-RUN wget https://github.com/jmespath/jp/releases/download/0.1.2/jp-linux-amd64 -qO /usr/local/bin/jp && chmod +x /usr/local/bin/jp
-
+# jq - we include jq as a useful tool
+RUN apk add --no-cache bash openssh ca-certificates jq wget openssl git \
+ && apk add --no-cache --virtual .build-deps gcc make openssl-dev libffi-dev musl-dev \
+ && update-ca-certificates \
+ && wget https://github.com/jmespath/jp/releases/download/${JP_LATEST_VERSION}/jp-linux-amd64 -qO /usr/local/bin/jp \
+ && chmod +x /usr/local/bin/jp \
+# pip wheel - required for CLI packaging
+# jmespath-terminal - we include jpterm as a useful tool
+ && pip install --no-cache-dir --upgrade jmespath-terminal -r requirements.txt \
```

Comment:
No need to install requirements.txt.

Reply:
This could by my python noobness, but I though the intention of the `requirements.txt` file was similar to  `packages.config` for nuget.

If almost all python projects follow this approach, why do this one differently?